tests/test_views.py

```diff
@@ -4582,6 +4528,246 @@ def test_confirm_user_signal_called_when_user_confirms_email(self):
+            assert_true(node.is_pending_registration)
+
+    @mock.patch('framework.tasks.handlers.enqueue_task')
+    def test_register_draft_registration_with_embargo_creates_embargo(self, mock_enquque):
+        url = self.node.api_url_for('register_draft_registration', draft_id=self.draft._id)
+        self.app.post_json(
+            url,
+            {
+                'registrationChoice': 'embargo',
+                'embargoEndDate': "Fri, 01 Jan {year} 05:00:00 GMT".format(year=str(dt.date.today().year + 1))
+            },
+            auth=self.auth)
+
+        self.node.reload()
+        # Most recent node is a registration
+        reg = Node.load(self.node.node__registrations[-1])
+        assert_true(reg.is_registration)
+        # The registration created is not public
+        assert_false(reg.is_public)
+        # The registration is pending an embargo that has not been approved
+        assert_true(reg.is_pending_embargo)
+        assert_false(reg.embargo_end_date)
+
+    @mock.patch('framework.tasks.handlers.enqueue_task')
+    def test_register_draft_registration_with_embargo_adds_to_parent_project_logs(self, mock_enquque):
+        initial_project_logs = len(self.node.logs)
+        self.app.post(
+            self.node.api_url_for('register_draft_registration', draft_id=self.draft._id),
+            self.valid_embargo_payload,
+            content_type='application/json',
+            auth=self.user.auth
+        )
+        self.node.reload()
+        # Logs: Created, registered, embargo initiated
+        assert_equal(len(self.node.logs), initial_project_logs + 1)
+
+    @mock.patch('framework.tasks.handlers.enqueue_task')
+    def test_register_draft_registration_with_embargo_is_not_public(self, mock_enqueue):
+        res = self.app.post(
+            self.node.api_url_for('register_draft_registration', draft_id=self.draft._id),
+            self.valid_embargo_payload,
+            content_type='application/json',
+            auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 202)
+
+        registration = Node.find().sort('-registered_date')[0]
+
+        assert_true(registration.is_registration)
+        assert_false(registration.is_public)
+        assert_true(registration.is_pending_embargo)
+        assert_is_not_none(registration.embargo)
+
+    @mock.patch('framework.tasks.handlers.enqueue_task')
+    def test_register_draft_registration_invalid_embargo_end_date_raises_HTTPError(self, mock_enqueue):
+        res = self.app.post(
+            self.node.api_url_for('register_draft_registration', draft_id=self.draft._id),
+            self.invalid_embargo_date_payload,
+            content_type='application/json',
+            auth=self.user.auth,
+            expect_errors=True
+        )
+
+        assert_equal(res.status_code, 400)
+
+    def test_get_draft_registrations_gets_drafts_for_node(self):
+
+        dummy = NodeFactory()
+        for i in range(5):
```

Comment:
@samchrisinger what is the purpose of this initial loop? Are we solely making sure that drafts that aren't actually appended to the drafts list are not returned by `DraftRegistration#get_draft_registrations`?